python/activity_stream/widget_note.py

```diff
@@ -383,16 +508,107 @@ def set_note_info(self, data):
         # higher level than this widget, but we're still going to be
         # careful here, because we saw this bug crop up during Cut Support
         # QA.
-        self.ui.content.setText(data.get("content", ""))
+        self._set_note_content_text(data.get("content", ""))
+        # The underlying text document's line count is not showing the correct
+        # number immediately after setting the document's text directly. Trigger a refresh
+        # of the widget size with the updated correct line count.
+        QtCore.QTimer.singleShot(0, self._update_note_content_size)
 
         # This allows selections from higher-level layouts to occur. If
         # we don't set this, the label accepts and blocks mouse click
         # events, which means if you expect to select the note widget
         # itself you can't click on the note contents.
-        self.ui.content.setAttribute(QtCore.Qt.WA_TransparentForMouseEvents)
-        
+        self.ui.content_editable.setAttribute(QtCore.Qt.WA_TransparentForMouseEvents)
+
         # format note links
         if self.show_note_links:
             html_link_box_data = data.get("note_links", []) + data.get("tasks", [])
             links_html = self.__generate_note_links_table(html_link_box_data)
             self.ui.links.setText(links_html)
+            self.ui.links.show()
+
+    ##########################################################################
+    # internal utilities
+
+    def _set_note_content_text(self, text):
+        """
+        Sets the UI content of the note text widget and stores a copy for state management.
+        """
+        self._saved_content = text
+        self.ui.content_editable.document().setPlainText(text)
+
+    def _update_note_content_size(self):
+        """
+        Resizes the Note text widget to match the number of visible lines up to the max.
+        """
+        widget = self.ui.content_editable
+        viewable_lines = widget.document().lineCount()
+        visible_lines = viewable_lines if viewable_lines <= self.MAX_VISIBLE_LINES else self.MAX_VISIBLE_LINES
+        # Adding a constant here for now, this FixedHeight results in the text touching
+        # the bottom of the text frame. Needs investigation.
+        widget.setFixedHeight(widget.fontMetrics().height() * visible_lines + 10)
```

Comment:
We need to QA this in various DCCs and on different OSes before this can be released.